crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -314,6 +322,12 @@ impl<'w, 's, 'a> BuildChildren for EntityCommands<'w, 's, 'a> {
         self.commands().add(AddChild { child, parent });
         self
     }
+
+    fn set_parent(&mut self, parent: Entity) -> &mut Self {
+        let child = self.id();
+        self.commands().add(AddChild { child, parent });
```

Comment:
Does that really work? What if there's already the opposite `AddChild` from the above `add_child()` call, do we correctly overwrite? I'm a bit surprised that we don't need to do any more book-keeping here.

Reply:
I'd like to see a simple test :)